zerver/lib/markdown/fenced_code.py

```diff
@@ -237,27 +237,60 @@ def done(self) -> None:
         self.processor.pop()
 
 
+class InnerHandler(BaseHandler):
```

Comment:
So, I'm a bit reluctant about this change, because this file is a forked copy of python-markdown upstream's file of the same name, and rewriting it with a different pattern means it'd be more or less impossible to rebase on top of upstream's bug fixes.  Maybe that's already the case, and/or this isn't significant (because mostly the things you're refactoring are new classes we added ourselves)?   May be worth a look at upstream for this question.

Reply:
Yes, it's already the case. We're only using the preprocessor (in FencedBlockPreprocessor) class of Python-markdown. Python markdown doesn't even allow to not close fence, but in Zulip, we're autoclosing fences.

Reply:
Yes, I'm only refactoring the classes we already have because there is a lot of repetition in code, and also, to fix #17610 I need all the handler(except the OuterHandler) classes to have one function.